torch/csrc/distributed/c10d/FlightRecorder.cpp

```diff
@@ -159,18 +159,31 @@ void DebugInfoWriter::write(const std::string& trace) {
 
   // Check if the file was opened successfully.
   if (!file.is_open()) {
-    LOG(ERROR) << "Error opening file for writing NCCLPG debug info: "
+    LOG(ERROR) << "Error opening file for writing Flight Recorder debug info: "
                << filename_;
     return;
   }
 
-  file.write(trace.data(), static_cast<std::streamsize>(trace.size()));
-  if (!file) {
-    LOG(ERROR) << "Error opening file for writing NCCLPG debug info: "
+  if (!file.write(trace.data(), static_cast<std::streamsize>(trace.size()))) {
+    if (file.bad()) {
+      LOG(ERROR) << "Serious error writing Flight Recorder debug info to file: "
+                 << filename_;
+    } else {
+      LOG(ERROR) << "Error writing Flight Recorder debug info to file: "
+                 << filename_;
+    }
```

Comment:
Curious, what is the distinction between "serious error" and error?

Reply:
I just just leave it as a single error - less code. Let me clean this up.
From the documentation - bad usually means something is seriously wrong with the file.  
https://cplusplus.com/reference/ios/ios/bad/